### PR TITLE
{AzureBlobStorage} fixes Azure/azure-sdk-for-net#32910 fix description of AccessTier Property

### LINKED ISSUE #32910: [BUG] Docs should clarify that AccessTier for V1 storage accounts is null
### Library name and version

Azure.Storage.Blobs 12.14.1

### Describe the bug

Blobs that are stored in V1 storage accounts don't have an access tier. This is all fine and well, until you forget about this and try to get the blob properties to get the access tier, don't expect a null value and try to convert it to an AccessTier type via the implicit string operator.

### Expected behavior

Either make the BlobProperties.AccessTier property nullable, or at least clarify in the documentation that the value can be null for V1 storage accounts.

The first option is the best because nullables exist for a reason, and will allow for code analysis to pick this up.
The second option is good enough if the first one is not feasible, because then it's at least clarified in the documentation.

### Actual behavior

The BlobProperties.AccessTier property is not marked as nullable, nor is it clarified that the value can be null.

Documentation currently states:
> The tier of page blob on a premium storage account or tier of block blob on blob storage LRS accounts. For a list of allowed premium page blob tiers, see [https://docs.microsoft.com/en-us/azure/virtual-machines/windows/premium-storage#features](https://learn.microsoft.com/en-us/azure/virtual-machines/windows/premium-storage#features). For blob storage LRS accounts, valid values are Hot/Cool/Archive.

### Reproduction Steps

Have a V1 storage account, and run the following snippet with the BlobClient referencing a blob in that storage account.

```cs
using Azure.Storage.Blobs;

var blob = new BlobClient(...);
var properties = await blob.GetPropertiesAsync();

Console.WriteLine(properties.Value.AccessTier); // is null

AccessTier accessTier = properties.Value.AccessTier; // ArgumentNullException
```

### Environment

_No response_

### PR DESCRIPTION
**Pull request by navba-MSFT**
fix description of AccessTier Property

fixes Azure/azure-sdk-for-net#32910 

The description of fix description of AccessTier Property should indicate the account types which support access tiers.

To avoid customers using GPv1 storage accounts to set the blob access tiers.

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs/src/Models/BlobProperties.cs [hunk 1]
```diff
@@ -206,8 +206,8 @@ public partial class BlobProperties
         /// <summary>
         /// The tier of page blob on a premium storage account or tier of block blob on blob storage LRS accounts.
         /// For a list of allowed premium page blob tiers, see
-        /// https://docs.microsoft.com/en-us/azure/virtual-machines/windows/premium-storage#features. For blob
-        /// storage LRS accounts, valid values are Hot/Cool/Archive.
+        /// https://docs.microsoft.com/en-us/azure/virtual-machines/windows/premium-storage#features. For general
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.Blobs/src/Models/BlobProperties.cs:209]: I think if you remove the trailing space after `general`. The CI will start running green.

**Comment by navba-MSFT** [comment]
@amnguye @seanmcc-msft The tests are now successfully passed. Could you please review and merge this once you get a chance ? Thanks in advance.

File: sdk/storage/Azure.Storage.Blobs/src/Models/BlobProperties.cs [hunk 2]
```diff
@@ -206,8 +206,8 @@ public partial class BlobProperties
...
+        /// purpose v2 and blob storage account types, the valid values are Hot/Cool/Archive.
```
Comment (jsquire) [inline sdk/storage/Azure.Storage.Blobs/src/Models/BlobProperties.cs:210]:
```suggestion
        /// <see href="https://docs.microsoft.com/azure/virtual-machines/windows/premium-storage#features" />. For general
        /// purpose v2 and blob storage account types, the valid values are:
        /// <list>
        ///  <item><description>Hot</description></item>
        ///  <item><description>Cool</description></item>
        ///  <item><description>Archive</description></item>
        /// </list>
```

Comment (jsquire) [reply sdk/storage/Azure.Storage.Blobs/src/Models/BlobProperties.cs:210]: If nothing else, please be sure to strip the language slug (`en/us`) from the URL.

**Review by amnguye — APPROVED** [review]

**Pull request merged**